--- a/arch/arm/Makefile
+++ b/arch/arm/Makefile
@@ -232,6 +232,7 @@
 machdirs-$(CONFIG_ARCH_FOOTBRIDGE)	+= arch/arm/mach-footbridge
 machdirs-$(CONFIG_ARCH_RPC)		+= arch/arm/mach-rpc
 machdirs-$(CONFIG_ARCH_SA1100)		+= arch/arm/mach-sa1100
+machdirs-$(CONFIG_ARCH_ALPINE)		+= arch/arm/mach-alpine
 KBUILD_CPPFLAGS += $(patsubst %,-I$(srctree)/%/include,$(machdirs-y))
 
 # The byte offset of the kernel image in RAM from the start of RAM.
--- a/arch/arm/mach-alpine/Kconfig
+++ b/arch/arm/mach-alpine/Kconfig
@@ -6,6 +6,7 @@
 	select ARM_AMBA
 	select ARM_GIC
 	select GENERIC_IRQ_CHIP
+	select ARM_HWCC_FLAG
 	select HAVE_ARM_ARCH_TIMER
 	select MFD_SYSCON
 	select FORCE_PCI
