--- a/drivers/pci/quirks.c
+++ b/drivers/pci/quirks.c
@@ -3209,6 +3209,9 @@ DECLARE_PCI_FIXUP_FINAL(PCI_VENDOR_ID_BROADCOM,
 DECLARE_PCI_FIXUP_FINAL(PCI_VENDOR_ID_BROADCOM,
 			PCI_DEVICE_ID_TIGON3_5715S,
 			quirk_msi_intx_disable_bug);
+DECLARE_PCI_FIXUP_FINAL(PCI_VENDOR_ID_AMAZON_ANNAPURNA_LABS,
+			PCI_DEVICE_ID_AMAZON_ANNAPURNA_LABS_RAID_DMA,
+			quirk_msi_intx_disable_bug);
 
 DECLARE_PCI_FIXUP_FINAL(PCI_VENDOR_ID_ATI, 0x4390,
 			quirk_msi_intx_disable_ati_bug);
