--- al_init_eth_kr.c	2024-08-31 06:56:34.003242100 -0400
+++ al_init_eth_kr.fixed.c	2024-08-31 11:01:54.923840300 -0400
@@ -394,11 +394,11 @@
 {
 	switch (coef) {
 	case AL_ETH_KR_COEF_C_MINUS:
-		return report->c_minus;
+		return (enum al_eth_kr_cl72_coef_op)(report->c_minus);
 	case AL_ETH_KR_COEF_C_PLUS:
-		return report->c_plus;
+		return (enum al_eth_kr_cl72_coef_op)(report->c_plus);
 	case AL_ETH_KR_COEF_C_ZERO:
-		return report->c_zero;
+		return (enum al_eth_kr_cl72_coef_op)(report->c_zero);
 	}
 
 	return 0;
